### PR TITLE
CLDR-16066 fix LDML 42 bad links found by W3C checkLink

### PR DESCRIPTION
**Pull request by pedberg-icu**
CLDR-16066

- [x] This PR completes the ticket.

FIx LDML 42 spec links (broken or inefficient) per report from W3C checkLink tool (https://validator.w3.org/checklink).

Some of these are just replacing a URL with what it already redirects to, if it is the same or nearest possible result. Many of these are links into our github repo, changing https://github.com/unicode-org/cldr/tree/.... to https://github.com/unicode-org/cldr/blob/...

### CODE REVIEW

File: docs/ldml/tr35-collation.md [hunk 1]
```diff
@@ -547,7 +547,7 @@ A collation type name that starts with "private-", for example, "private-kana",
 
 > 👉 **Note**: There is an on-line demonstration of collation at [[LocaleExplorer](tr35.md#LocaleExplorer)] that uses the same rule syntax. (Pick the locale and scroll to "Collation Rules", near the end.)
 
-> 👉 **Note**: In CLDR 23 and before, LDML collation files used an XML format. Starting with CLDR 24, the XML collation syntax is deprecated and no longer used. See the _[CLDR 23 version of this document](https://www.unicode.org/reports/tr35/tr35-31/tr35-collation.md#Collation_Tailorings)_ for details about the XML collation syntax.
+> 👉 **Note**: In CLDR 23 and before, LDML collation files used an XML format. Starting with CLDR 24, the XML collation syntax is deprecated and no longer used. See the _[CLDR 23 version of this document](https://www.unicode.org/reports/tr35/tr35-31/tr35-collation.html#Collation_Tailorings)_ for details about the XML collation syntax.
```
Comment (pedberg-icu) [inline docs/ldml/tr35-collation.md:550]: This should be treated like an external (network) URL and hence html, not an internal reference using md

File: docs/ldml/tr35.md [hunk 2]
```diff
@@ -1340,7 +1340,7 @@ Note that the language of locale data may differ from the language of localized
 
 #### 3.10.2 <a name="Hybrid_Locale" href="#Hybrid_Locale">Hybrid Locale Identifiers</a>
 
-Hybrid locales have intermixed content from 2 (or more) languages, often with one language's grammatical structure applied to words in another. These are commonly referred to with portmanteau words such as _Franglais, [​Spanglish](https://en.oxforddictionaries.com/definition/spanglish)_ or _Denglish_. Hybrid locales do not _not_ reference text simply containing two languages: a book of parallel text containing English and French, such as the following, is not Franglais:
+Hybrid locales have intermixed content from 2 (or more) languages, often with one language's grammatical structure applied to words in another. These are commonly referred to with portmanteau words such as _Franglais, [​Spanglish](https://en.wikipedia.org/wiki/Spanglish)_ or _Denglish_. Hybrid locales do not _not_ reference text simply containing two languages: a book of parallel text containing English and French, such as the following, is not Franglais:
```
Comment (pedberg-icu) [inline docs/ldml/tr35.md:1343]: Oxford dictionaries has gone to using a paywall, with a different URL

**Review by srl295 — APPROVED** [review]

File: docs/ldml/tr35.md [hunk 3]
```diff
@@ -3589,7 +3589,7 @@ To canonicalize a given _source_:
 4. Transform _source_ according to that rule
 5. loop (goto #3)
 
-## Processing LocaleIds
+### <a name="processing-localeids" href="#processing-localeids">Processing LocaleIds</a>
```
Comment (pedberg-icu) [inline docs/ldml/tr35.md:3592]: This and some following headers were at the wrong level

File: docs/ldml/tr35-collation.md (hunk 1 above)
Comment (srl295) [reply docs/ldml/tr35-collation.md:550]: i thought i broke and fixed these, but may have missed some

File: docs/ldml/tr35.md (hunk 2 above)
Comment (srl295) [reply docs/ldml/tr35.md:1343]: ok

File: docs/ldml/tr35.md (hunk 3 above)
Comment (srl295) [reply docs/ldml/tr35.md:3592]: was the fix-tocs tool used to regenerate this or done manually?

the ToC is updated automatically before html is generated, but should be run locally also

**Review by rickmcgowan — COMMENTED** [review]
Mostly look reasonable to me. I don't know anything about the "tree" versus "blob" things in Github, however.

File: docs/ldml/tr35.md (hunk 3 above)
Comment (pedberg-icu) [reply docs/ldml/tr35.md:3592]: For Appendix C, both the TOC and the section heads were at the wrong level for some sections, so TOC generation would not fix it. Also the link target and href were missing for all of the Appendix XC subsections (though present in the TOC somehow).

**Pull request merged**